demo/src/main/java/io/paymenthighway/demo/AddCardActivity.kt

```diff
@@ -81,11 +110,24 @@ class AddCardActivity : AppCompatActivity(), ValidationListener {
     override fun isValidDidChange(isValid: Boolean) {
         mAddCardButton.isEnabled = isValid
         if (isValid) {
-            val imm = this.getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
-            imm.hideSoftInputFromWindow(currentFocus.windowToken, 0)
+            hideKeyboard()
         }
     }
 
+    override fun imageDrawable(type: EditTextType): Int = getDrawableId(type)
+
+    private fun hideKeyboard() {
+        if (currentFocus == null) return
+        val imm = this.getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
+        imm.hideSoftInputFromWindow(currentFocus!!.windowToken, 0)
+    }
+
+    private fun getDrawableId(type: EditTextType): Int = when (type) {
```

Comment:
Was this related to some limitations? Feels a bit weird having to care about theme colors in the code.
Or is it necessary for dynamic theme changes perhaps?